### DISCUSSION TITLE
main: 5afa2249 is a breaking change: ts_configs (nvim-treesitter.configs) is gone

### DISCUSSION THREAD

**Discussion by wookayin** [nvim-treesitter/nvim-treesitter#4776] [General]
The latest commit on the 'main' (5afa2249) seems to be an intentional breaking change, but I believe many configs are using `ts_configs` (even README): https://github.com/nvim-treesitter/nvim-treesitter#modules

```
require'nvim-treesitter.configs'.setup {
...
}
```

which will start to break when merged. I use the latest master/main branch to be adventurous. I was guessing that this might be a part of v1.0.

Is this really a necessary (breaking) change? Can we have this documented clearly at least, or can we have some backward compatibility (e.g., having some deprecation warning when the old module is used)? Neither #4767, the commit message, nor any issue have mentioned such an incompatible change.

I can see that if this was intentional all users have to migrate and catch up the team's decision, but any help or feedback would be appreciated. /cc @clason

**Comment by wookayin** [comment]
Looking at #4767 this is a step toward v1.0 so users should be prepared for an inevitable breaking change.

**Comment by clason** [comment]
Yes, the `main` branch is full of breaking changes; it's in effect a new plugin and not meant to be merged into `master`. **DO NOT USE THIS BRANCH YET**.
